src/hotspot/share/jvmci/jvmciRuntime.cpp

```diff
@@ -904,19 +958,232 @@ static void _flush_log() {
 
 // Function for shared library JavaVM to exit HotSpot on a fatal error
 static void _fatal() {
+  Thread* thread = Thread::current_or_null_safe();
+  if (thread != NULL && thread->is_Java_thread()) {
+    JavaThread* jthread = (JavaThread*) thread;
+    JVMCIRuntime* runtime = jthread->libjvmci_runtime();
+    if (runtime != NULL) {
+      int javaVM_id = runtime->get_shared_library_javavm_id();
+      fatal("Fatal error in JVMCI shared library JavaVM[%d] owned by JVMCI runtime %d", javaVM_id, runtime->id());
+    }
+  }
   intx current_thread_id = os::current_thread_id();
   fatal("thread " INTX_FORMAT ": Fatal error in JVMCI shared library", current_thread_id);
 }
 
-JNIEnv* JVMCIRuntime::init_shared_library_javavm() {
-  JavaVM* javaVM = (JavaVM*) _shared_library_javavm;
-  if (javaVM == NULL) {
+JVMCIRuntime::JVMCIRuntime(JVMCIRuntime* next, int id, bool for_compile_broker) {
+  _init_state = uninitialized;
```

Comment:
Why is this constructor using assignments rather than a ctor-initializer?  There's lots of old code that doesn't, and the Style Guide doesn't require it (yet - https://bugs.openjdk.java.net/browse/JDK-8252586), but that's the trend (and generally considered best practice).